read.go

```diff
@@ -226,24 +230,22 @@ func readNativeFrames(d dicomio.Reader, parsedData *Dataset, fc chan<- *frame.Fr
 		buf := make([]int, int(pixelsPerFrame)*samplesPerPixel)
 		for pixel := 0; pixel < int(pixelsPerFrame); pixel++ {
 			for value := 0; value < samplesPerPixel; value++ {
-				if bitsAllocated == 8 {
-					val, err := d.ReadUInt8()
-					if err != nil {
-						return nil, bytesRead, err
+
+				n, err := d.Read(pixelBuf)
```

Comment:
Thought about this a little more the second time around (sorry for not bringing this up earlier). Looking at the `io.Reader` [interface](https://golang.org/pkg/io/#Reader) I _think_ a single call to read may read "up to len(p)" bytes. Perhaps we should use `io.ReadFull` to try to read _at least_ the size of the buffer.

In reality for values like 16 or 32 bits, calls to `Read` may still work no problem, but it may be more "correct" to use something like `io.ReadFull` which won't be expensive in the case that `Read` would've worked on the first try. This may also insulate us from the exact implementation of the reader under the hood we're reading from, since there is no garuntee that it must fill the buffer on the first call to read.

WDYT?

```suggestion
				n, err := io.ReadFull(d, pixelBuf)
```

Reply:
Good catch. `Read(buffer)` can in fact return less than `n` bytes when reading from a network socket, for example.